indexdata.go

```diff
@@ -157,25 +157,29 @@ func (d *indexData) getChecksum(idx uint32) []byte {
 	return d.checksums[start : start+crc64.Size]
 }
 
-// TODO (stefan): Update this function to return meaningful stats for compound
-// shards.
-func (d *indexData) calculateStatsForRepoIndex(i int) RepoListEntry {
+// calculates stats for files in the range [start, end). It is assumed that all
+// files in the range belong to the same repository.
+func (d *indexData) calculateStatsForFileRange(start, end uint32) RepoListEntry {
```

Comment:
for the merge process we need to check the invariant that repos are over contiguous ranges. Should we just represent it that way, instead of keeping the invariant?